runtime/src/bank.rs

```diff
@@ -3046,6 +3036,8 @@ impl Bank {
         let partition_width = (PREFIX_MAX - partition_count + 1) / partition_count + 1;
         let mut start_key_prefix = if start_index == 0 && end_index == 0 {
             0
+        } else if start_index + 1 == partition_count && is_fix_enabled {
```

Comment:
@carllin Hmm, I think you don't need gating for this because we can be proved not using overflowing `start_key_prefix`. Could you proof-read my following reasoning?:

1. When this `if` clause's condition ( `start_index + 1 == partition_count`) is met, `end_index` is `>= start_index` because of this assert `assert!(start_index <= end_index);` at line 3017.
2. Then, `end_key_prefix` is always set to `PREFIX_MAX` because `end_index + 1 == partition_count` should be always evaluated to true at line 3045 as well.
3. Also note that there is no case of `end_index + 1 > partition_count` because of `start_index < partition_count` and `end_index < partition_count` at lines 3018-3019
4. Finally, if `end_key_prefix == PREFIX`, we're overwriting `start_key_prefix` with `end_key_prefix` at line 3055.
5. So, there should be no observable case to use overflowing value of `start_key_prefix`.

Maybe, you're only encountered this bug with debug build. In that case, it'll immediately panic on the detection of overflowing value..

Reply:
in other words, this is harmless unless a debug build validator is running on any live clusters, which isn't practice because of slow binary to begin with... ;)

Reply:
@ryoqun yeah, I think I agree. For 4, the code in question:

```
if start_index != 0 && start_index == end_index {
            // n..=n (n != 0): a noop pair across epochs without a gap under
            // multi_epoch_cycle, just nullify it.
            if end_key_prefix == PREFIX_MAX {
                start_key_prefix = end_key_prefix;
                start_pubkey = end_pubkey;
```

Then the only case `end_key_prefix != PREFIX` is if `start_index == 0`. But if `start_index ==0` and `start_index + 1 == partition_count`,  then `partition_count == 1`, and so b/c `end_index < partition_count`, then `end_index == 0`. 

This would mean the: `let mut start_key_prefix = if start_index == 0 && end_index == 0 {
            0
        }`
clause would execute, bypassing my newly added `else if`, so behavior is also unchanged.

Thanks for the detailed reasoning, I should have looked at it more carefully, thanks!

Reply:
no problem! well, I haven't thought enough when writing the code, that's my bad to begin with... ;)